kahuna/public/js/components/gu-lazy-gallery/gu-lazy-gallery.js

```diff
@@ -0,0 +1,144 @@
+import angular from 'angular';
+import Rx from 'rx';
+import 'rx-dom';
+
+import template from './gu-lazy-gallery.html!text';
+
+import '../../util/rx';
+
+export var lazyGallery = angular.module('gu.lazyGallery', ['util.rx']);
+
+function asInt(string) {
+    return parseInt(string, 10);
+}
+
+lazyGallery.controller('GuLazyGalleryCtrl', ['$scope', 'subscribe$', function($scope, subscribe$) {
+    let ctrl = this;
+    ctrl.canGoNext = false;
+    ctrl.canGoPrev = false;
+
+    ctrl.init = function({items$, preloadedItems$, currentIndex$}) {
+        const itemsCount$ = items$.map(items => items.length).distinctUntilChanged();
+
+        const buttonCommands$ = new Rx.BehaviorSubject('galleryStart');
```

Comment:
I think this is fine, but I'm wondering: if you did it the other way (where you assigned all the handlers within the creation block) could you avoid using BehaviorSubject? 

I'm asking because I was just looking at the docs and behavior subject seems to be for "[a value that changes over time](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/subjects/behaviorsubject.md)" which I don't think really applies to the buttonpress stream. 

just something to be aware of